Compiler warning on Linux due to unused result (-Wunused-result)

Error showed up here:

https://chromium-review.googlesource.com/c/angle/angle/+/2601340

https://ci.chromium.org/ui/p/angle/builders/try/linux-clang-rel/18555/overview

```
../../third_party/vulkan-deps/vulkan-loader/src/loader/loader.c:2659:9: error: ignoring return value of function declared with 'warn_unused_result' attribute [-Werror,-Wunused-result]
        fread(buffer, 1, sizeof(buffer), file);
        ^~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Code: https://github.com/KhronosGroup/Vulkan-Loader/blob/294480cb7f3e5b18082ad0294c61bb7648d00820/loader/loader.c#L2659

@mark-lunarg is this warning enabled in the loader build? Should we fix the specific warning and/or enable it in the CI or should I remove this warning from the supported set in GN?

--- a/loader/loader.c
+++ b/loader/loader.c
@@ -2653,11 +2653,11 @@ static VkResult loader_get_json(const struct loader_instance *inst, const char *
         goto out;
     }
     // NOTE: We can't just use fseek(file, 0, SEEK_END) because that isn't guaranteed to be supported on all systems
+    size_t fread_ret_count = 0;
     do {
-        // We're just seeking the end of the file, so this buffer is never used
         char buffer[256];
-        fread(buffer, 1, sizeof(buffer), file);
-    } while (!feof(file));
+        fread_ret_count = fread(buffer, 1, 256, file);
+    } while (fread_ret_count == 256 && !feof(file));
     len = ftell(file);
     fseek(file, 0, SEEK_SET);
     json_buf = (char *)loader_stack_alloc(len + 1);

